ee/tabby-webserver/src/oauth/mod.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use axum::{
+    extract::{Query, State},
+    http::StatusCode,
+    response::Redirect,
+    routing, Router,
+};
+use serde::Deserialize;
+use tracing::error;
+
+use crate::{
+    oauth::github::GithubClient,
+    schema::{
+        auth::{AuthenticationService, GithubAuthError},
+        ServiceLocator,
+    },
+};
+
+pub mod github;
+
+#[non_exhaustive]
+struct OAuthState {
```

Comment:
since all fields of this struct are already `Arc`, I suggest just derive `Clone` for the struct